### PR TITLE
Merge Config Fixes

### PR DESCRIPTION
**Pull request by CyberiumShadow**
### Proposed Semver Increment Bump: [MAJOR/MINOR/PATCH]
N/A
### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Fixes Pedall's issue as reported in Discord.
- Object.assign used to deep-merge objects in the default options to the user options.
-

### (If Applicable) What Issue does it fix?

 Fixes... keys containing Objects not being merged properly.

### CODE REVIEW

**Comment by CyberiumShadow** [comment]
@UnseenFaith If you feel like the two `Object.assign`'s can be moved elsewhere, do let me know.

**Comment by CyberiumShadow** [comment]
This also has the benefit of fixing an issue Turtle had...

**Comment by JohnDoesCodes** [comment]
I think you forgot to add a semicolon on line 65 and 82 eh?
`Object.assign(options.disabled, this.DEFAULT_OPTIONS.disabled, Object.assign({}, options.disabled));`
`Object.assign(options.provider, this.DEFAULT_OPTIONS.provider, Object.assign({}, options.provider));`

**Comment by CyberiumShadow** [comment]
@JohnDoesCodes Thanks for pointing that out haha

File: src/functions/mergeConfig.js [hunk 1]
```diff
@@ -18,8 +18,13 @@ const defaultPermStructure = new PermLevels()
 
 
 module.exports = (options) => {
-  for (const key in this.DEFAULT_OPTIONS) { // eslint-disable-line
+    for (const key in this.DEFAULT_OPTIONS) { // eslint-disable-line
```
Comment (UnseenFaith) [inline src/functions/mergeConfig.js]: fixerino space por favor

**Pull request merged**